src/core/providers/ogr/qgsogrprovidermetadata.cpp

```diff
@@ -668,6 +667,21 @@ bool QgsOgrProviderMetadata::saveLayerMetadata( const QString &uri, const QgsLay
       // file based, but not a geopackage -- store as .qmd sidecar file instead
       // (possibly there's other formats outside of GPKG which also has some native means of storing metadata,
       // which could be added for those formats before we resort to the sidecar approach!)
+
+      QString adjustedPath = path;
+      if ( adjustedPath.endsWith( QLatin1String( ".gz" ), Qt::CaseInsensitive ) )
+      {
+        adjustedPath.chop( 3 );
+      }
+      else if ( adjustedPath.endsWith( QLatin1String( ".zip" ), Qt::CaseInsensitive ) || adjustedPath.endsWith( QLatin1String( ".tar" ), Qt::CaseInsensitive ) || adjustedPath.endsWith( QLatin1String( ".tgz" ), Qt::CaseInsensitive ) )
+      {
+        adjustedPath.chop( 4 );
+      }
+      else if ( adjustedPath.endsWith( QLatin1String( ".tar.gz" ), Qt::CaseInsensitive ) )
+      {
+        adjustedPath.chop( 7 );
+      }
```

Comment:
couldn't that be rewritten as something like:
```
for( const QString& ending : { QLatin1String(".gz"), QLatin1String( ".zip" ), etc .. }
{
     if(  adjustedPath.endsWith( ending, Qt::CaseInsensitive ) )
      {
        adjustedPath.chop( ending.size() );
        break;
      }
}
```

Reply:
@rouault , good idea, implemented.